cli/src/main/java/org/jboss/as/cli/impl/ModelControllerClientFactory.java

```diff
@@ -50,39 +51,19 @@
         void handleClose();
     }
 
-    ModelControllerClient getClient(ControllerAddress address, CallbackHandler handler,
+    ModelControllerClient getClient(ControllerAddress address, CallbackHandler handler, Set<CallbackKind> callbackKinds,
             boolean disableLocalAuth, SecurityFactory<SSLContext> sslContextFactory, boolean fallbackSslContext, int connectionTimeout,
             ConnectionCloseHandler closeHandler, ProtocolTimeoutHandler timeoutHandler, String clientBindAddress) throws IOException;
 
-    ModelControllerClientFactory DEFAULT = new ModelControllerClientFactory() {
```

Comment:
I agree with the removal of this static instance. Its added value is very low and the signature would become even more confusing with the introduction of CallbackKind. 
Even if one is relying on it, it would have to recompile and re-emplementing it directly is straight forward.
BTW, ModelControllerClientConfiguration.Builder requires a change to support CallbackKind.